### PR TITLE
feat: adding locale to word module

### PR DESCRIPTION
**Pull request by gustavobastian**
In this PR, the word module has incorporate the locale feature. All previous functions (ej, faker::word::sample()) can receive a locale parameter(it is necessary to pass the length argument, ej: sample(7,faker::locale::en_US)).

### CODE REVIEW

File: include/faker-cxx/word.h [hunk 1]
```diff
@@ -23,12 +27,30 @@ namespace faker::word
  * faker::word::sample(5) // "spell"
  * @endcode
  */
-FAKER_CXX_EXPORT std::string_view sample(std::optional<unsigned> length = std::nullopt);
+FAKER_CXX_EXPORT std::string_view sample(std::optional<unsigned> length = std::nullopt,const faker::Locale locale = Locale::en_US);
```
Comment (cieslarmichal) [inline include/faker-cxx/word.h:30]: No need for faker prefix as we are in faker namespace, also const is not required as its copy anyway

File: tests/modules/string_test.cpp [hunk 2]
```diff
@@ -196,14 +196,14 @@ TEST_F(StringTest, shouldGenerateSampleStringWithGuarantee2)
 
 TEST_F(StringTest, shouldGenerateSampleStringWithGuarantee3)
 {
-    const auto sampleLength{20};
+    const auto sampleLocaleength{20};
     const GuaranteeMap guarantee = {{'(', {0, 4}}, {'{', {0, 2}}, {'\\', {0, 1}}, {'/', {0, 5}}};
```
Comment (cieslarmichal) [inline tests/modules/string_test.cpp:200]: Typo everywhere in leength

File: tests/modules/word_test.cpp [hunk 3]
```diff
@@ -276,3 +277,333 @@ TEST_F(WordTest, returnsFirstElementWhenNoLengthMatch)
 
     ASSERT_TRUE(result == "three");
 }
+class WordTestLocale : public TestWithParam<Locale>
+{
```
Comment (cieslarmichal) [inline tests/modules/word_test.cpp:281]: Please use parametrized tests with TEST_P and use locale param within test like in book module tests

File: CMakeLists.txt [hunk 4]
```diff
@@ -8,7 +8,7 @@ project(faker-cxx
 include(cmake/CompilerWarnings.cmake)
 include(CMakeDependentOption)
 
-option(USE_SYSTEM_DEPENDENCIES "Use fmt and GTest from system" OFF)
+option(USE_SYSTEM_DEPENDENCIES "Use fmt and GTest from system" ON)
 option(BUILD_EXAMPLES "Build examples" OFF)
```
[the remark this answers is not in the document]
Comment (gustavobastian) [reply CMakeLists.txt:12]: yes... fixing that.

File: include/faker-cxx/word.h (hunk 1 above)
Comment (gustavobastian) [reply include/faker-cxx/word.h:30]: fixed.

File: src/modules/internet.cpp [hunk 5]
```diff
@@ -341,11 +341,11 @@ std::string anonymousUsername(unsigned maxLength)
     else if (maxLength > defaultMax)
         maxLength = defaultMax;
 
-    const std::integral auto adjectiveLength = number::integer<unsigned>(3, 1 + maxLength / 2);
+    const std::integral auto adjectiveLocaleength = number::integer<unsigned>(3, 1 + maxLength / 2);
```
Comment (cieslarmichal) [inline src/modules/internet.cpp:344]: Revert

File: tests/modules/string_test.cpp (hunk 2 above)
Comment (gustavobastian) [reply tests/modules/string_test.cpp:200]: done

File: include/faker-cxx/word.h [hunk 6]
```diff
@@ -23,12 +27,30 @@ namespace faker::word
...
+FAKER_CXX_EXPORT std::string_view sample(std::optional<unsigned> length = std::nullopt,const Locale locale = Locale::en_US);
+
+
+/**
+ * @brief Returns a random .
+ *
+ * @param length The expected length of the .
+ * If 0,it will return a random . 
+ * @param locale The locale. Defaults to `Locale::en_US`. 
+ * @returns Random sample word.
```
Comment (cieslarmichal) [inline include/faker-cxx/word.h:39]: something broke here `Returns a random .`, `will return a random .`

File: include/faker-cxx/word.h [hunk 7]
```diff
@@ -23,12 +27,30 @@ namespace faker::word
...
+ *
+ * @code
+ * faker::word::sampleLocale(0) // "protection"
+ * faker::word::sampleLocale(5,Locale::es_AR) // "casas"
+ * @endcode
+ */
+
+FAKER_CXX_EXPORT std::string_view sampleLocale(unsigned length = 0,const Locale locale = Locale::en_US);
```
Comment (cieslarmichal) [inline include/faker-cxx/word.h:47]: why there are two functions? sample and sampleLocale? it should be only sample, applies to other functions as well

File: tests/modules/word_test.cpp [hunk 8]
```diff
@@ -276,3 +303,407 @@ TEST_F(WordTest, returnsFirstElementWhenNoLengthMatch)
...
+    auto idiomsMapLocal=getIdiomsMap(locale); 
+    const auto generatedAdjective = adjective(5,locale);
+
+    ASSERT_TRUE(std::ranges::any_of(idiomsMapLocal.adjetives, [generatedAdjective](const std::string_view& word)
+                                    { return word == generatedAdjective; }));
+}
+
+TEST_P(WordTestLocale, shouldGenerateAdjectiveWithNonExistingLength)
+{
+    Locale locale = GetParam();
```
Comment (cieslarmichal) [inline tests/modules/word_test.cpp:341]: could be const auto instead of type, applies to other tests as well

File: tests/modules/word_test.cpp [hunk 9]
```diff
@@ -276,3 +303,407 @@ TEST_F(WordTest, returnsFirstElementWhenNoLengthMatch)
...
+    auto idiomsMapLocal=getIdiomsMap(locale);
```
Comment (cieslarmichal) [inline tests/modules/word_test.cpp:342]: I think it could be const

File: tests/modules/word_test.cpp [hunk 10]
```diff
@@ -276,3 +303,407 @@ TEST_F(WordTest, returnsFirstElementWhenNoLengthMatch)
...
+TEST_F(WordTestLocale, shouldReturnFrenchSampleifAskedforFrenchWordUsingSample)
+{
+    faker::Locale locale= faker::Locale::fr_FR;
+    const auto generatedSample = sample(0,locale);
+    auto dataset=_allWords_map.at(locale);
+    ASSERT_TRUE(std::ranges::any_of(dataset, [generatedSample](const std::string_view& word)
+                                    { return word == generatedSample; }));
+}
+
+
```
Comment (cieslarmichal) [inline tests/modules/word_test.cpp:707]: please remove extra new lines (double new lines)

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

File: include/faker-cxx/word.h (hunk 7 above)
Comment (gustavobastian) [reply include/faker-cxx/word.h]: solved

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**